resource/hlapi/bindings/c++/reapi_cli_impl.hpp

```diff
@@ -89,6 +93,423 @@ void reapi_cli_t::clear_err_message ()
+        tmp_err +=  ": ERROR: can't set subsystem\n";
+        throw std::runtime_error (tmp_err);
+    }
+
+    if ( !(fgraph = create_filtered_graph ())) {
+        tmp_err = __FUNCTION__;
+        tmp_err +=  ": ERROR: can't create filtered graph\n";
+        throw std::runtime_error (tmp_err);
+    }
+
+    jobid_counter = 1;
+    if (params.prune_filters != ""
+        && matcher->set_pruning_types_w_spec (
+                                matcher->dom_subsystem (),
+                                params.prune_filters)
+                                < 0) {
+        tmp_err = __FUNCTION__;
+        tmp_err +=  ": ERROR: can't initialize pruning filters\n";
+        throw std::runtime_error (tmp_err);
+    }
+
+    if (traverser->initialize (fgraph, db, matcher) != 0) {
+        tmp_err = __FUNCTION__;
+        tmp_err +=  ": ERROR: can't initialize traverser\n";
+        throw std::runtime_error (tmp_err);
+    }
+
+    format = match_writers_factory_t::get_writers_type (params.match_format);
+    if ( !(writers = match_writers_factory_t::create (format))) {
+        tmp_err = __FUNCTION__;
+        tmp_err +=  ": ERROR: can't create match writer\n";
+        throw std::runtime_error (tmp_err);
+    }
+
+    return;
+}
+
+const std::string &resource_query_t::get_resource_query_err_msg () const
+{
+    return m_err_msg;
+}
+
+const std::string &resource_query_t::get_traverser_err_msg () const
+{
+    return traverser->err_message ();
+}
+
+const bool resource_query_t::job_exists (const uint64_t jobid)
+{
+    return jobs.find (jobid) != jobs.end ();
+}
+
+const uint64_t resource_query_t::get_job_counter () const
+{
+    return jobid_counter;
+}
+
+const std::shared_ptr<job_info_t> &resource_query_t::get_job (const uint64_t jobid)
+{
+    return jobs.at (jobid);
+}
+
+const bool resource_query_t::allocation_exists (const uint64_t jobid)
+{
+    return allocations.find (jobid) != allocations.end ();
+}
+
+const bool resource_query_t::reservation_exists (const uint64_t jobid)
+{
+    return reservations.find (jobid) != reservations.end ();
+}
+
+void resource_query_t::clear_resource_query_err_msg ()
+{
+    m_err_msg = "";
+}
+
+void resource_query_t::clear_traverser_err_msg ()
+{
+    traverser->clear_err_message ();
+}
+
+void resource_query_t::set_reservation (const uint64_t jobid)
+{
+    reservations[jobid] = jobid;
+}
+
+void resource_query_t::erase_reservation (const uint64_t jobid)
+{
+    reservations.erase (jobid);
+}
+
+void resource_query_t::set_allocation (const uint64_t jobid)
+{
+    allocations[jobid] = jobid;
+}
+
+void resource_query_t::erase_allocation (const uint64_t jobid)
+{
+    allocations.erase (jobid);
+}
+
+void resource_query_t::set_job (const uint64_t jobid,
+                                const std::shared_ptr<job_info_t> &job)
+{
+    jobs[jobid] = job;
+}
+
+int resource_query_t::remove_job (const uint64_t jobid)
+{
+    int rc = -1;
+    if ((rc = traverser->remove ((int64_t)jobid)) == 0) {
```

Comment:
Minor issue: traverser class uses `int64_t` for jobid and for a very rare case, this conversion can be a problem. I'd suggest:

```suggestion
    if (jobid > std::numeric_limits<int64_t>::max) {
        errno = EOVERFLOW;
        return rc;
    }
    if ((rc = traverser->remove (static_cast<int64_t>(jobid)) == 0) {

Reply:
Great catch, thanks!

Reply:
I'm going to leave this line as is:
```C++
    if ((rc = traverser->remove ((int64_t)jobid)) == 0) {
```
since changing it to:
```C++
    if ((rc = traverser->remove (static_cast<int64_t>(jobid)) == 0) {
```
results in a compilation error:
```sh
error: init-statement in selection statements only available with ‘-std=c++17’ or ‘-std=gnu++17
```

Reply:
Ugg. Alternatively, you can

```c++
rc = traverser->remove (static_cast<int64_t>(jobid);
if ( rc == 0) {
```

Reply:
Ok, it's fixed if I pull the `rc` assignment out of the `if` statement.